ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java

```diff
@@ -73,9 +73,21 @@ private void generateCommands() throws IOException, TemplateException {
         }
 
         if (ConverterUtil.hasContent(spec.getTypes())) {
+            // Generate EnumConstant Types, which Element types can be dependant on
+            // TODO Determine if dependency matters at code generation time... probably not.
             for (final Ds3Type typeEntry : spec.getTypes().values()) {
-                generateType(typeEntry);
-                generateTypeMatcher(typeEntry);
+                if (typeEntry.getElements().isEmpty()) {
```

Comment:
You may want to change this to checking if enum constants has content, since there can exist both enum constants and elements within a type.  At least within the Java generation, I discussed it with Ryan and he determined that when both exist within the same type, the elements are actually descriptive of internal variables that are not related to code generation.  Therefore, if both exist, generate an enum.  Not sure if its the same for the C generated code